runtime/vm/jfr.cpp

```diff
@@ -635,17 +647,106 @@ tearDownJFR(J9JavaVM *vm)
 	(*vmHooks)->J9HookUnregister(vmHooks, J9HOOK_VM_INITIALIZED, jfrVMInitialized, NULL);
 
 	/* Free global data */
+	freeConstantEvents(vm);
+
 	j9mem_free_memory((void*)vm->jfrBuffer.bufferStart);
 	memset(&vm->jfrBuffer, 0, sizeof(vm->jfrBuffer));
 	if (NULL != vm->jfrBufferMutex) {
 		omrthread_monitor_destroy(vm->jfrBufferMutex);
 		vm->jfrBufferMutex = NULL;
 	}
+	if (NULL != vm->jfrState.isConstantEventsInitializedMutex) {
+		omrthread_monitor_destroy(vm->jfrState.isConstantEventsInitializedMutex);
+		vm->jfrState.isConstantEventsInitializedMutex = NULL;
+	}
 	j9mem_free_memory(vm->jfrState.metaDataBlobFile);
 	vm->jfrState.metaDataBlobFile = NULL;
 	vm->jfrState.metaDataBlobFileSize = 0;
 }
 
+/**
+ * Helper to get the value of a system property.
+ *
+ * @param vm[in] the J9JavaVM
+ * @param propName[in] the system property name
+ */
+static const char *getSystemProp(J9JavaVM *vm, const char *propName) {
+	J9VMSystemProperty *jvmInfoProperty = NULL;
+	const char *value = "";
+	UDATA getPropertyResult = vm->internalVMFunctions->getSystemProperty(vm, propName, &jvmInfoProperty);
+	if (J9SYSPROP_ERROR_NOT_FOUND != getPropertyResult) {
+		value = jvmInfoProperty->value;
+	}
+	return value;
+}
+
+/**
+ * Initialize constantEvents.
+ *
+ * @param vm[in] the J9JavaVM
+ */
+static void
+initConstantEvents(J9JavaVM *vm)
+{
+
+	PORT_ACCESS_FROM_JAVAVM(vm);
```

Comment:
JVMInfo events should be in its own function because in the future youll be adding more constantEvents